legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-core/src/main/java/org/finos/legend/engine/persistence/components/relational/api/RelationalIngestorAbstract.java

```diff
@@ -540,6 +572,43 @@ private void acquireLock()
         }
     }
 
+    private List<IngestorResult> verifyIfRequestAlreadyProcessedPreviously(SchemaEvolutionResult schemaEvolutionResult)
+    {
+        List<IngestorResult> result = new ArrayList<>();
+        if (enableConcurrentSafety() && enableIdempotencyCheck())
+        {
+            LOGGER.info("Perform Idempotency Check");
+            LogicalPlan logicalPlan = LogicalPlanFactory.getLogicalPlanForIdempotencyCheck(
+                    enrichedDatasets.metadataDataset().orElseThrow(IllegalStateException::new),
+                    ingestRequestId().orElseThrow(IllegalStateException::new),
+                    enrichedDatasets.mainDataset().datasetReference().name().orElseThrow(IllegalStateException::new));
+            SqlPlan physicalPlan = transformer.generatePhysicalPlan(logicalPlan);
+            List<TabularData> tabularDataList = executor.executePhysicalPlanAndGetResults(physicalPlan);
+            MetadataDataset metadataDataset = enrichedDatasets.metadataDataset().get();
+            if (!tabularDataList.isEmpty())
+            {
+                List<Map<String, Object>> metadataResults = tabularDataList.get(0).getData();
+                for (Map<String, Object> metadata: metadataResults)
+                {
+                    Timestamp ingestionTimestampUTC = (Timestamp) metadata.get(metadataDataset.batchStartTimeField());
+                    String batchStatus = (String) metadata.get(metadataDataset.batchStatusField());
+                    batchStatus = batchStatus.equalsIgnoreCase(batchSuccessStatusValue())  ? IngestStatus.SUCCEEDED.name() : batchStatus;
+                    IngestorResult ingestorResult = IngestorResult.builder()
+                            .batchId((int) metadata.get(metadataDataset.tableBatchIdField()))
+                            .putAllStatisticByName(readValueAsMap((String) metadata.get(metadataDataset.batchStatisticsField())))
```

Comment:
We may have to do a String.valueOf(), as this casting may cause error in DuckDB and Postgres.
In the Postgres and DuckDB PR, we have made a such a change in a similar place: https://github.com/finos/legend-engine/blob/293388680d5088f37d31a04c3dcb52461f628d68/legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-core/src/main/java/org/finos/legend/engine/persistence/components/relational/api/ApiUtils.java#L232

Reply:
Done